Lib/fontParts/base/font.py

```diff
@@ -540,264 +693,390 @@ def kerningSortKeyFunc(pair):
+        """Get the native font's temporary lib object.
+
+        This is the environment implementation
+        of :attr:`BaseFont.tempLib`.
+
+        :return: A temporary instance of a :class:`BaseLib` subclass.
+        :raises NotImplementedError: If the method has not been
+            overridden by a subclass.
+
+        .. important::
+
+            Subclasses must override this method.
+
         """
         self.raiseNotImplementedError()
 
     # -----------------
     # Layer Interaction
     # -----------------
 
-    layers = dynamicProperty(
+    layers: dynamicProperty = dynamicProperty(
         "base_layers",
-        """
-        The font's :class:`BaseLayer` objects.
+        """Get the font's layer objects.
+
+        :return: A :class:`tuple` containing instances of
+            the :class:`BaseLayer` class.
+
+        Example::
 
             >>> for layer in font.layers:
             ...     layer.name
             "My Layer 1"
             "My Layer 2"
+
         """
     )
 
-    def _get_base_layers(self):
-        layers = self._get_layers()
+    def _get_base_layers(self) -> Tuple[BaseLayer, ...]:
+        layers: Tuple[BaseLayer, ...] = self._get_layers()
         for layer in layers:
             self._setFontInLayer(layer)
         return tuple(layers)
 
-    def _get_layers(self, **kwargs):
-        """
+    def _get_layers(self, **kwargs: Any) -> Tuple[BaseLayer, ...]:
+        r"""Get the native font's layer objects.
+
         This is the environment implementation of
-        :attr:`BaseFont.layers`. This must return an
-        :ref:`type-immutable-list` containing
-        instances of :class:`BaseLayer` subclasses.
-        The items in the list should be in the order
-        defined by :attr:`BaseFont.layerOrder`.
+        :attr:`BaseFont.layers`.
+
+        :param \**kwargs: Additional keyword arguments.
+        :return: A :class:`tuple` containing instances
+            of the :class:`BaseLayer` subclass. The items  should be in
+            the order defined by :attr:`BaseFont.layerOrder`.
+
+        .. important::
+
+            Subclasses must override this method.
 
-        Subclasses must override this method.
         """
         self.raiseNotImplementedError()
 
     # order
 
-    layerOrder = dynamicProperty(
+    layerOrder: dynamicProperty = dynamicProperty(
         "base_layerOrder",
-        """
-        A list of layer names indicating order of the layers in the font.
+        """Get or set the order of the layers in the font.
+
+        The value must be a :class:`list` or :class:`tuple` of layer
+        names as :class:`str` reflecting the desired order of the
+        font's :class:`BaseLayer` objects.
+
+        :return: A :class:`tuple` of layer names as :class:`str` in
+            their defined order.
+
+        Example::
 
             >>> font.layerOrder = ["My Layer 2", "My Layer 1"]
             >>> font.layerOrder
             ["My Layer 2", "My Layer 1"]
```

Comment:
Should be `("My Layer 2", "My Layer 1")`

Reply:
Fixed.